userlib.go

```diff
@@ -342,7 +342,17 @@ func hmacEval(key []byte, msg []byte) ([]byte, error) {
 
 var HMACEval = hmacEval
 
-var HashKDF = hmacEval
+// HashKDF (uses the same algorithm as hmacEval)
+func HashKDF(key []byte, msg []byte) ([]byte, error) {
+	if len(key) != 16 && len(key) != 24 && len(key) != 32 {
+		panic(errors.New("The input as key for HashKDF should be a 16-byte key."))
+	}
+
+	mac := hmac.New(sha512.New, key)
+	mac.Write(msg)
+	res := mac.Sum(nil)
+	return res, nil
+}
```

Comment:
To match everything else, we should probably split this to a `hashKDF` function and then just export it using `var HashKDF = hashKDF`.

Reply:
Just pushed

Reply:
I’m going to re-push over the branch just to match the comment styles and whatnot.